test/e2e/storage/testsuites/fsgroupchangepolicy.go

```diff
@@ -0,0 +1,358 @@
+		if pattern.BindingMode != storagev1.VolumeBindingImmediate {
+			e2eskipper.Skipf("Test does not support non-immediate volume binding mode - skipping")
+		}
+		if pattern.VolType != testpatterns.DynamicPV {
+			e2eskipper.Skipf("Test does not support non Dynamic provisioned PV - skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		testVolumeSizeRange := s.GetTestSuiteInfo().SupportedSizeRange
+		resource := CreateVolumeResource(l.driver, l.config, pattern, testVolumeSizeRange)
+		l.resources = append(l.resources, resource)
+		TestPodFsGroupChangePolicyOnRootMismatch(l.config.Framework, l.cs, l.ns.Name, l.config.ClientNodeSelection, resource.Pvc)
+	})
+
+	ginkgo.It("pod fsgroup change policy (OnRootMismatch) for dynamic provisioned delayed volume binding", func() {
+		if pattern.BindingMode != storagev1.VolumeBindingWaitForFirstConsumer {
+			e2eskipper.Skipf("Test does not support non-delayed volume binding mode - skipping")
+		}
+		if pattern.VolType != testpatterns.DynamicPV {
+			e2eskipper.Skipf("Test does not support non Dynamic provisioned PV - skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		testVolumeSizeRange := s.GetTestSuiteInfo().SupportedSizeRange
+		resource := CreateVolumeResource(l.driver, l.config, pattern, testVolumeSizeRange)
+		l.resources = append(l.resources, resource)
+		TestPodFsGroupChangePolicyOnRootMismatch(l.config.Framework, l.cs, l.ns.Name, l.config.ClientNodeSelection, resource.Pvc)
+	})
+}
+
+// TestPodFsGroupChangePolicyAlways tests 'Always' policy.
+func TestPodFsGroupChangePolicyAlways(f *framework.Framework, cs clientset.Interface,
+	ns string, nodeSelection e2epod.NodeSelection, pvc *v1.PersistentVolumeClaim) {
+	gid1 := 2000
+	alwaysPodFsGroupPolicy := v1.PodFSGroupChangePolicy("Always")
+	podConfig := e2epod.Config{
+		NS:                     ns,
+		NodeSelection:          nodeSelection,
+		PVCs:                   []*v1.PersistentVolumeClaim{pvc},
+		FsGroup:                utilpointer.Int64Ptr(int64(gid1)),
+		PodFSGroupChangePolicy: &alwaysPodFsGroupPolicy,
+	}
+
+	pod, err := e2epod.CreateSecPodWithNodeSelection(cs, &podConfig, framework.PodStartTimeout)
+	framework.ExpectNoError(err)
+	framework.Logf("First Pod %s/%s started successfully", pod.Namespace, pod.Name)
+	cmd := "ls -l /mnt/volume1"
+	stdout, stderr, err := storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+	framework.Logf("pod exec for cmd %s, stdout: %s, stderr: %s", cmd, stdout, stderr)
+
+	cmd = "touch /mnt/volume1/1.txt"
+	_, _, err = storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+
+	storageutils.VerifyFilePathGidInPod(f, "/mnt/volume1/1.txt", strconv.Itoa(gid1), pod)
+
+	cmd = "mkdir /mnt/volume1/subdir"
+	_, _, err = storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+
+	cmd = "touch /mnt/volume1/subdir/2.txt"
+	_, _, err = storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+
+	storageutils.VerifyFilePathGidInPod(f, "/mnt/volume1/subdir/2.txt", strconv.Itoa(gid1), pod)
```

Comment:
No way to check set group id? Or do you think it's not necessary with the subpath check.

Reply:
since a new file 2.txt inherited the gid, it indicates the gid bit is set.

Reply:
Yeah I was just wondering if we wanted to test the setgid bit directly. Maybe that's overly fussy though.